src/components/atoms/LatLonFromLocationSearch.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { ChangeEvent, Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { MetAPIVariables } from '../../queries/metApi';
+import { Button, Combobox, TextInputField, toaster } from 'evergreen-ui';
+import { Address } from '../../types/Address';
+import debounce from 'lodash.debounce';
+
+type LatLonInputSetProps = {
+    state: MetAPIVariables;
+    setState: Dispatch<SetStateAction<MetAPIVariables>>;
+};
+
+const LatLonFromLocationSearch: React.FC<LatLonInputSetProps> = (props) => {
+    const [state, setState] = useState<{ lat: string; lon: string }>({
+        lat: String(props.state.lat),
+        lon: String(props.state.lon),
+    });
+
+    const [locations, setLocations] = useState<
+        {
+            label: string;
+            lat: number;
+            lon: number;
+        }[]
+    >([]);
+
+    // To avoid problems with difficult digit entry for user
+    useEffect(() => {
+        props.setState({ lat: Number(state.lat), lon: Number(state.lon) });
+    }, [state]);
+
+    const update = (key: keyof typeof state) => (e: ChangeEvent<HTMLInputElement>) =>
+        setState((prevState) => ({ ...prevState, [key]: e.target.value }));
+
+    const setToMyLocation = () => {
+        navigator.geolocation.getCurrentPosition(
+            (position) => {
+                setState({ lat: String(position.coords.latitude), lon: String(position.coords.longitude) });
+            },
+            () => {
+                toaster.warning('Du må tillate lokasjonstjenester for at denne tjenesten skal fungere.');
+            },
+        );
+    };
+
+    const onInputChange = async (e: React.ChangeEvent<HTMLInputElement>): Promise<void> => {
+        const locationData = await fetch(
+            `https://ws.geonorge.no/adresser/v1/sok?sok=${e.target.value}&treffPerSide=10`,
+        ).then((res) => res.json());
```

Comment:
```suggestion
        const rawLocationData = await fetch(
            `https://ws.geonorge.no/adresser/v1/sok?sok=${e.target.value}&treffPerSide=10`,
        );
        const locationData: Array<Address> = await rawLocationData.json();
```

Use either Promises or async/await.

Reply:
ok...ay?